egg/core/interaction.py

```diff
@@ -2,23 +2,46 @@
 
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-
+import random
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
 from typing import Dict, Iterable, Optional
 
 import torch
 import torch.distributed as distrib
 
 
-@dataclass
-class LoggingStrategy:
-    store_sender_input: bool = True
-    store_receiver_input: bool = True
-    store_labels: bool = True
-    store_message: bool = True
-    store_receiver_output: bool = True
-    store_message_length: bool = True
+class BaseStrategy(ABC):
```

Comment:
We probably don't need an hierarchy with abstract classes when we actually have one non-abstract class? Let's wait when/if it is needed, as this day might never come.

This also results in a somewhat large change & complexity for the sake of adding one float parameter?

Reply:
Noted. I still think some kind of skeleton should be provided, but let's wait for now.
Removed in bc782e2